plugins/BEdita/API/src/Controller/MediaController.php

```diff
@@ -0,0 +1,98 @@
+<?php
+/**
+ * BEdita, API-first content management framework
+ * Copyright 2018 ChannelWeb Srl, Chialab Srl
+ *
+ * This file is part of BEdita: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * See LICENSE.LGPL or <http://gnu.org/licenses/lgpl-3.0.html> for more details.
+ */
+
+namespace BEdita\API\Controller;
+
+use BEdita\Core\Filesystem\Thumbnail;
+use BEdita\Core\Model\Entity\Stream;
+use Cake\Database\Expression\QueryExpression;
+use Cake\Network\Exception\BadRequestException;
+
+/**
+ * Controller for media.
+ *
+ * @since 4.0.0
+ */
+class MediaController extends ObjectsController
+{
+
+    /**
+     * Get and validate IDs in request URL.
+     *
+     * @return int[]
+     */
+    protected function getIds()
+    {
+        $id = $this->request->getParam('id');
+        $ids = $this->request->getQuery('ids');
```

Comment:
is there a limit on `'ids'` number? 
should not exceed pagination limit...

Reply:
Right, I'm adding this check.

Reply:
Actually there's the option of paginating the results.

E.g.: `GET /media/thumbs?ids=1,2,3,4&page_size=2` would return two pages of results, assuming each of those Media has one associated Stream. Query parameters for pagination would be the same as usual.

Which do you prefer? Paginated results or **400 Bad Request** if `count($ids) > Configure::read('Pagination.maxLimit')`?

Reply:
I'd say `400 Bad Request`

Reply:
:+1 on  `400 Bad Request`